### ISSUE TITLE
Add an enumeratee to support parallel processing

### ISSUE THREAD

**Issue by travisbrown** [travisbrown/iteratee#114]
It would be nice to have something like `gatherMap` that would allow us to run a transformation in parallel over batches of items. I can imagine a method like this:

```scala
def gatherMap[F[_]: Monad, O, I]
  (gather: Vector[F[I]] => F[Vector[I]], size: Int)(f: O => F[I]):: Enumeratee[F, O, I]
```

This would be a little nicer if we had a `NonDeterminism` type class, and I guess we could introduce our own, but passing in a `gather` function isn't too bad, either. I'd imagine the implementation would be pretty simple with `sequenceI`, but I haven't written it out yet.
